syntaxes/omg.idl.tmLanguage.json

```diff
@@ -0,0 +1,567 @@
+				{"include": "#type_dcl"},
+				{"include": "#const_dcl"},
+				{"include": "#value_dcl"}
+			]
+		},
+		"const_dcl": {
+			"name": "meta.declaration.const.omg.idl",
+			"begin": "(const)\\s+(\\w+)(\\:\\:\\w+)*\\s+(\\w+)\\s*=",
+			"end": ";",
+			"beginCaptures": {
+				"1": { "name": "storage.modifier.const.omg.idl"},
+				"2": {
+					"patterns": [
+						{"include": "#const_type"}
+					]		
+				},
+				"3": {},
+				"4": {"name": "entity.name.omg.idl"}
+			},
+			"patterns": [
+				{
+					"name": "constant.numeric.omg.idl",
+					"match": "\\d+"	
+				},
+				{"include": "#string_literal"}
+			]
+		},
+		"const_type": {
+			"patterns": [
+				{"include": "#integer_type"},
+				{"include": "#floating_pt_type"},
+				{"include": "#fixed_pt_const_type"},
+				{"include": "#char_type"},
+				{"include": "#wide_char_type"},
+				{"include": "#boolean_type"},
+				{"include": "#octet_type"},
+				{"include": "#string_type"},
+				{"include": "#string_type_unbounded"}
+			]
+		},
+		"literal": {
+			"patterns": [
+				{"include": "#integer_literal"},
+				{"include": "#floating_pt_literal"},
+				{"include": "#fixed_pt_literal"},
+				{"include": "#character_literal"},
+				{"include": "#wide_character_literal"},
+				{"include": "#boolean_literal"},
+				{"include": "#string_literal"}
+			]
+		},
+		"integer_literal": {
+			"patterns": [{
+				"name": "constant.numeric.omg.idl",
+				"match": "\\b(\\d+|0x\\h+)\\b"
+			}]
+		},
+		"boolean_literal": {
+			"patterns": [{
+				"name": "constant.language.omg.idl",
+				"match": "\\b(TRUE|FALSE)\\b"
+			}]
+		},
+		"positive_int_const": {
+			"patterns": [{
+				"name": "constant.numeric.omg.idl",
+				"match": "(\\d+)"
+			}]
+		},
+		"string_literal": {
+			"name": "string.quoted.double.omg.idl",
+			"begin": "\"",
+			"end": "\"",
+			"patterns": [
+				{
+					"name": "constant.character.escape.omg.idl",
+					"match": "\\\\."
+				}
+			]
+		},
+		"type_dcl": {
+			"patterns": [
+				{"include": "#constr_type_dcl"},
+				{"include": "#typedef_dcl"}
+			]
+		},
+		"type_spec": {
+			"patterns": [
+				{"include": "#simple_type_spec"}
+			]
+		},
+		"simple_type_spec": {
+			"patterns": [
+				{"include": "#base_type_spec"},
+				{"include": "#template_type_spec"}
+			]
+		},
+		"base_type_spec": {
+			"patterns": [
+				{"include": "#integer_type"},
+				{"include": "#floating_pt_type"},
+				{"include": "#char_type"},
+				{"include": "#wide_char_type"},
+				{"include": "#boolean_type"},
+				{"include": "#octet_type"}
+			]
+		},
+		"floating_pt_type": {
+			"patterns": [
+				{
+					"name": "storage.type.primitive.omg.idl",
+					"match": "\\b(float|double|long double)\\b"
+				}
+			]
+		},
+		"integer_type": {
+			"patterns": [
+				{"include": "#signed_int"},
+				{"include": "#unsigned_int"}
+			]
+		},
+		"signed_int": {
+			"patterns": [
+				{"include": "#signed_short_int"},
+				{"include": "#signed_long_int"},
+				{"include": "#signed_longlong_int"}
+			]
```

Comment:
IDL 4.2 has introduced support for 8-bit integer values. They are definde in BNF as follows:
```
(208) <signed_tiny_int> ::= “int8”
(209) <unsigned_tiny_int> ::= “uint8”
```
Following your notation, you could add an include for `signed_tiny_int` under `signed_int` and `unsigned_tiny_int` under `unsigned_int`.